code/game/objects/items/devices/swapper.dm

```diff
@@ -112,3 +112,6 @@
 		if(ismob(B))
 			var/mob/M = B
 			to_chat(M, "<span class='warning'>[linked_swapper] activates, and you find yourself somewhere else.</span>")
+			//victim ckey is appended which is weird and idk, but I left this here for now.
+			//log_combat(B, "swapped to [AREACOORD(B)], swapper activated by [user.ckey]")
```

Comment:
```suggestion
			log_combat(B, user, "was swapped to [AREACOORD(B)]")
```
Try something like that and see if that doesnt log it twice if you swap yourself.